test/e2e/install_test.go

```diff
@@ -27,62 +28,62 @@ var _ = Describe("Operator Install", func() {
 		operatorName string
 		operator     *operatorv1alpha1.Operator
 	)
+	When("An operator is installed from an operator catalog", func() {
+		BeforeEach(func() {
+			ctx = context.Background()
+			pkgName = "argocd-operator"
+			operatorName = fmt.Sprintf("operator-%s", rand.String(8))
+			operator = &operatorv1alpha1.Operator{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: operatorName,
+				},
+				Spec: operatorv1alpha1.OperatorSpec{
+					PackageName: pkgName,
+				},
+			}
+			operatorCatalog := &catalogd.CatalogSource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-catalog",
+				},
+				Spec: catalogd.CatalogSourceSpec{
+					Image: "quay.io/operatorhubio/catalog:latest",
+				},
+			}
+			err := c.Create(ctx, operatorCatalog)
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(func(g Gomega) {
+				err = c.Get(ctx, types.NamespacedName{Name: "test-catalog"}, operatorCatalog)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(len(operatorCatalog.Status.Conditions)).To(Equal(1))
+				g.Expect(operatorCatalog.Status.Conditions[0].Message).To(Equal("catalog contents have been unpacked and are available on cluster"))
+			}).WithTimeout(2 * time.Minute).WithPolling(defaultPoll).Should(Succeed())
+		})
+		It("resolves the specified package with correct bundle path", func() {
+			By("creating the Operator resource")
+			err := c.Create(ctx, operator)
+			Expect(err).ToNot(HaveOccurred())
 
-	var testNamespace string
-	cleanup := func() {}
-	ctx = context.TODO()
-	BeforeEach(func() {
-		testNamespace = createTestNamespace(ctx, c, "registry-grpc-")
-		cleanup = applyCRDifNotPresent(ctx)
-		testPrefix := "registry-grpc-"
+			By("eventually reporting a successful resolution and bundle path")
+			Eventually(func(g Gomega) {
+				err = c.Get(ctx, types.NamespacedName{Name: operator.Name}, operator)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(len(operator.Status.Conditions)).To(Equal(1))
+				g.Expect(operator.Status.Conditions[0].Message).To(Equal("install was successful"))
+			}).WithTimeout(defaultTimeout).WithPolling(defaultPoll).Should(Succeed())
```

Comment:
It it necessary to pass `g Gomega` to the function? I would expect us to be able to do something like this:
```go
Eventually(func() {
    Expect(...)
})
```

Reply:
Turns out: 

```
 [FAILED] The function passed to Eventually had an invalid signature of func().  Functions passed to Eventually must either:

  	(a) have return values or
  	(b) take a Gomega interface as their first argument and use that Gomega instance to make assertions.

  You can learn more at https://onsi.github.io/gomega/#eventually
```

Reply:
I know we're used to the first approach in our existing test suites, but if we want to discuss the merits of a) vs b) we can do that in a separate PR. For now I've reverted back to using b)

Reply:
Oh, that function needs to return something so it can be asserted against.

Reply:
Yea that's what I meant in my previous comment, that we're using the func signature you mentioned in other test suites but changing the current approach in the code base seems like a follow up PR